pkg/webhook/bootstrap.go

```diff
@@ -81,147 +97,71 @@ func (s *Server) setBootstrappingDefault() {
 	if s.Service != nil && s.Port != 443 {
 		s.Port = 443
 	}
-	if s.certProvisioner == nil {
-		s.certProvisioner = &cert.Provisioner{}
+
+	var certWriter writer.CertWriter
+	var err error
+	if s.Secret != nil {
+		certWriter, err = writer.NewSecretCertWriter(
+			writer.SecretCertWriterOptions{
+				Secret: s.Secret,
+			})
+	} else {
+		certWriter, err = writer.NewFSCertWriter(
+			writer.FSCertWriterOptions{
+				Path: s.CertDir,
+			})
+	}
+	if err != nil {
+		s.err = err
+		return
 	}
-	if s.CertGenerator == nil {
-		s.CertGenerator = &generator.SelfSignedCertGenerator{}
+	s.certProvisioner = &cert.Provisioner{
+		CertWriter: certWriter,
 	}
 	if s.Writer == nil {
 		s.Writer = os.Stdout
 	}
 }
 
-// bootstrap returns the configuration of admissionWebhookConfiguration in yaml format if dryrun is true.
+// bootstrap writes the configuration of admissionWebhookConfiguration in yaml format if dryrun is true.
 // Otherwise, it creates the the admissionWebhookConfiguration objects and service if any.
-// It also provisions the certificate for your admission server.
+// It also provisions the certificate for the admission server.
 func (s *Server) bootstrap(dryrun bool) error {
```

Comment:
May be we should rename this function to `installWebhookConfig` to make it more explicit. Also, can `dryrun` be a server option ?